Source/BugsnagClient.m

```diff
@@ -774,51 +829,46 @@ - (void)batteryChanged:(NSNotification *)notif {
                  toTabWithName:BSGKeyDeviceState];
 }
 
-- (void)orientationChanged:(NSNotification *)notif {
-    NSString *orientation;
-    UIDeviceOrientation deviceOrientation =
-        [UIDevice currentDevice].orientation;
-
-    switch (deviceOrientation) {
-    case UIDeviceOrientationPortraitUpsideDown:
-        orientation = @"portraitupsidedown";
-        break;
-    case UIDeviceOrientationPortrait:
-        orientation = @"portrait";
-        break;
-    case UIDeviceOrientationLandscapeRight:
-        orientation = @"landscaperight";
-        break;
-    case UIDeviceOrientationLandscapeLeft:
-        orientation = @"landscapeleft";
-        break;
-    case UIDeviceOrientationFaceUp:
-        orientation = @"faceup";
-        break;
-    case UIDeviceOrientationFaceDown:
-        orientation = @"facedown";
-        break;
-    default:
-        return; // always ignore unknown breadcrumbs
+/**
+ * Called when an orientation change notification is received to record an
+ * equivalent breadcrumb.
+ *
+ * @param notification The orientation-change notification
+ */
+- (void)orientationChanged:(NSNotification *)notification {
+    UIDeviceOrientation currentDeviceOrientation = [UIDevice currentDevice].orientation;
+    NSString *orientation = BSGOrientationNameFromEnum(currentDeviceOrientation);
+    
+    // Short-circuit the exit if we don't have enough info to record a full breadcrumb
+    // or the orientation hasn't changed (false positive).
+    if (!orientation) {
+        return;
+    }
+    else if (!_lastOrientation || [orientation isEqualToString:_lastOrientation]) {
+        _lastOrientation = orientation;
+        return;
     }
 
-    NSDictionary *lastBreadcrumb =
-        [[self.configuration.breadcrumbs arrayValue] lastObject];
-    NSString *orientationNotifName =
-        BSGBreadcrumbNameForNotificationName(notif.name);
+    NSDictionary *lastBreadcrumb = [[self.configuration.breadcrumbs arrayValue] lastObject];
```

Comment:
This feels error-prone - users could choose to reject the breadcrumb via an `OnBreadcrumb` callback. It should also be possible to just use `self.lastOrientation` to remove duplicate orientation events, unless I've missed some additional complexity here.

Reply:
Restructured the entire method after discussion.